core/index/src/main/java/mil/nga/giat/geowave/core/index/ByteArrayRange.java

```diff
@@ -108,26 +109,28 @@ public int compareTo(
 				other.getEnd());
 	}
 
-	public static final void mergeIntersections(
+	public static final List<ByteArrayRange> mergeIntersections(
 			List<ByteArrayRange> ranges,
 			int maxRanges ) {
```

Comment:
Yes.  I am torn on this. 
The algorithm needs to score range distance (between each other) and size, choosing the closest/shortest ranges first until the maxRange is met.  A priority heap ordered by a size/distance metric or a sorted list ordered (in reverse) for optimal removal is needed.  Both are O(nlogn).  If you have a O(n) solution, then lets fix this now.  Otherwise, I am tempted to just drop it.  Given some maxRange (e.g. 100), a worst case scenario is probably something like three or four times this amount.